tests/test_sources.py

```diff
@@ -240,6 +387,54 @@ def connected_fault(
     return Fault(planes)
 
 
+@given(
+    fault=st.builds(
+        connected_fault,
+        lengths=st.lists(st.floats(0.1, 100), min_size=1, max_size=10),
+        width=st.floats(0.1, 100),
+        strike=st.floats(0, 179),
+        start_coordinates=st.builds(
+            coordinate, lat=st.floats(-50, -31), lon=st.floats(160, 180)
+        ),
+    ),
+    distance=st.floats(1, 1000),
+)
+def test_fault_rjb(fault: Fault, distance: float):
+    # if dip dir is too close to strike it will create a degenerate geometry that rjb distance isn't designed for anyway.
+    buffer = shapely.buffer(fault.geometry, distance * 1000)
+    for point in coordinates.nztm_to_wgs_depth(np.array(buffer.exterior.coords)):
+        assert np.isclose(
+            fault.rjb_distance(point),
+            distance * 1000,
+            atol=1e-4,
+        )
+
+
+@given(
+    fault=st.builds(
+        connected_fault,
+        lengths=st.lists(st.floats(0.1, 100), min_size=1, max_size=10),
+        width=st.floats(0.1, 100),
+        strike=st.floats(0, 179),
+        start_coordinates=st.builds(
+            coordinate, lat=st.floats(-50, -31), lon=st.floats(160, 180)
+        ),
+    ),
+    point=st.builds(
+        coordinate,
+        lat=st.floats(-50, -31),
+        lon=st.floats(160, 180),
+        depth=st.floats(0, 100),
+    ),
+)
+def test_fault_rrup(fault: Fault, point: np.ndarray):
```

Comment:
The first case should be tested with hypothesis testing. The second case doesn't really make sense to test because it would require constructing an invalid plane in the first place and to me it's not the job of `rrup_distance` to check that, it's the job of the constructor.

Reply:
But I can add a test for the constructor for invalid planes, and a test for rrup distance within the plane.

Reply:
@joelridden See [the new PR changes](https://github.com/ucgmsim/source_modelling/pull/14/files#diff-5f2630533f745c6079637bcfdab384f3d08c41a5b14c36173b4dbc55423dc984R187) that checks that plane bounds have:

1. Two values with a minimum depth.
2. Two values with a maximum depth.
3. Four points total, spanning exactly a plane. Not a line or a point.
4. They are oriented correctly according to the docs.

These four points guarantee that the fail state of a degenerate line segment is never triggered and so testing for a value error on a constructed plane is not required. I have added tests for the above four points in the testing code on #14.